solver/llbsolver/solver.go

```diff
@@ -308,6 +250,105 @@ func (s *Solver) Solve(ctx context.Context, id string, sessionID string, req fro
 	}, nil
 }
 
+func runCacheExporters(ctx context.Context, exporters []RemoteCacheExporter, j *solver.Job, cached *result.Result[solver.CachedResult], inp *result.Result[cache.ImmutableRef]) (map[string]string, error) {
+	eg, ctx := errgroup.WithContext(ctx)
+	g := session.NewGroup(j.SessionID)
+	var cacheExporterResponse map[string]string
+	resps := make([]map[string]string, len(exporters))
+	for i, exp := range exporters {
+		func(exp RemoteCacheExporter, i int) {
```

Comment:
Like [cache imports](https://github.com/moby/buildkit/blob/f2c41e15cb4d725649efe47a8d3023d0b35fdbe8/solver/llbsolver/bridge.go#L388-L397) we should store a cache export ID so we make sure the ref (registry) / folder (local) / scope (gha) is dedup: https://github.com/moby/buildkit/blob/f2c41e15cb4d725649efe47a8d3023d0b35fdbe8/solver/llbsolver/bridge.go#L83

Reply:
I'm looking into that.

Reply:
@crazy-max added deduplication of cache exporters in [702ba77](https://github.com/moby/buildkit/pull/3024/commits/702ba77d4abdf9b2e47da99ef4a6b0d3249550be)